jupyterlab/labbox_ephys_widgets_jp/src/extensions/extensionInterface.ts

```diff
@@ -57,6 +59,65 @@ export interface MetricPlugin extends LabboxPlugin {
     
 }
 
+// Sorting curation
+export type SortingCuration = {
+    labelsByUnit: {[key: string]: string[]}
+    labelChoices: string[]
+}
+export const defaultSortingCuration: SortingCuration = {
+    labelChoices: ["accept", "reject", "mua", "artifact", "noise"].sort(),
+    labelsByUnit: {}
+}
+
+export type SortingCurationDispatch = (action: SortingCurationAction) => void
+
+type AddLabelSortingCurationAction = {
+    type: 'AddLabel',
+    unitId: number
+    label: string
+}
+
+type RemoveLabelSortingCurationAction = {
+    type: 'RemoveLabel',
+    unitId: number
+    label: string
+}
+
+export type SortingCurationAction = AddLabelSortingCurationAction | RemoveLabelSortingCurationAction
+
+export const sortingCurationReducer: Reducer<SortingCuration, SortingCurationAction> = (state: SortingCuration, action: SortingCurationAction): SortingCuration => {
```

Comment:
This reducer is used by the jupyter widget, but not by the web gui. That's because the web gui uses the global redux state to dispatch the curation actions.